### PR TITLE
EventListener: add support for indexing listings in Postgres and Elasticsearch

### PR DESCRIPTION
**Pull request by franckc**
First pull request? Read our [guide to contributing](http://docs.originprotocol.com/#contributing)

### Checklist:

- [ ] Code contains relevant tests for the problem you are solving
- [X] Ensure all new and existing tests pass
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)
- [X] Submit to the `develop` branch instead of `master`

### Description:
Submitting this PR to get some early feedback on the approach. 

This PR includes:
 - Basic support for the EventListener to index directly into Elasticsearch and Postgres rather than using a webhook.
 - Create separate npm module from origin-js under the daemon directory. This is a **temporary** solution until we move to a monorepo structure.

Note: this is a starting point and there is still lot of work remaining to make this production ready !
 - dynamically configure Postgres and ElasticSearch client
 - index Offer events
 - handle Listing state changes
 - add unit tests and ensure those get run by Travis CI
 - and probably lot more...

### CODE REVIEW

File: daemon/listener/listener.js [hunk 1]
```diff
@@ -185,13 +190,26 @@ async function handleLog(log, rule, contractVersion, context) {
...
+  if (context.config.search) {
+    await search.indexListing(listing.id, listing.ipfsData.data)
+  }
+
+  if (context.config.db) {
+    await db.insertListing(listing.id, listing.ipfsData.data)
+  }
 
   if (context.config.webhook) {
-    process.stdout.write('\n-- WEBHOOK to ' + context.config.webhook + ' --\n')
```
Comment (DanielVF) [inline daemon/listener/listener.js]: Good! Nice catch.

File: daemon/listener/listener.js [hunk 2]
```diff
@@ -185,13 +190,26 @@ async function handleLog(log, rule, contractVersion, context) {
...
+    console.log('\n-- WEBHOOK to ' + context.config.webhook + ' --\n')
     await postToWebhook(context.config.webhook, json)
-  } else {
```
Comment (DanielVF) [inline daemon/listener/listener.js]: I still think we need some way to send to stdout - it's really handy for debugging the listener, as well as debugging other systems, and it's just a handy feature to do unixy things with.

File: daemon/listener/listener.js [hunk 3]
```diff
@@ -298,6 +316,13 @@ process.argv.forEach(arg => {
 })
 
 const config = {
-  webhook: args['--webhook']
+  // Call webhook to process event.
+  webhook: args['--webhook'],
+  // Index events in the search index.
+  search: args['--search'],
```
Comment (DanielVF) [inline daemon/listener/listener.js]: "search" might be the wrong name for this - it sounds too much like an action. Like you want the listener to search something.

"elasticsearch" would be fine though.

File: daemon/listener/db.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+/*
+ * Returns all rows from the listing table.
+ * @throws Throws an error if the read operation failed.
+ * @returns A promise that resolves to a list of rows (can be empty).
+ *
+ * TODO(franck): add support for pagination.
+ */
+async function getListings() {
+  return pool.query(`SELECT * FROM ${listingTable}`, []).then((res) => {
```
Comment (DanielVF) [inline daemon/listener/db.js]: If you feel like it, you could use `await` in these instead of `then`. Might make things clearer and easier to read.

**Comment by DanielVF** [comment]
It's amazing how little code you had to add to do these changes.

File: daemon/listener/listener.js (hunk 3 above)
Comment (franckc) [reply daemon/listener/listener.js]: changed to elasticsearch

File: daemon/listener/listener.js (hunk 2 above)
Comment (franckc) [reply daemon/listener/listener.js]: Yes, agreed. The --verbose option will dump the log to stdout.

File: daemon/listener/db.js (hunk 4 above)
Comment (franckc) [reply daemon/listener/db.js]: Yes, good point. It is cleaner. Updated.

**Comment by sparrowDom** [comment]
LGTM

**Pull request merged**